lib/jsonschema/filesystem.go

```diff
@@ -0,0 +1,38 @@
+package jsonschema
+
+import (
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type FS struct {
+	raw fs.FS
+}
+
+// NewFS will typically be passed an embed.FS to easily load separate .json schema files
+func NewFS(raw fs.FS) *FS {
+	return &FS{raw: raw}
+}
+
+// Loads the given filename suffixing with ".json"
+func (f *FS) LoadJSONExt(filepath string) gojsonschema.JSONLoader {
+	return f.Load(fmt.Sprintf("%s.json", filepath))
```

Comment:
json schemas _have_ to be JSON, so I'm not sure there's much benefit to allowing this as an option - would just make it the only way